packages/replay-internal/test/integration/recordingMode.test.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * @vitest-environment jsdom
+ */
+
+import { beforeEach, describe, expect, test } from 'vitest';
+import type { Replay } from '../../src/integration';
+import type { ReplayContainer } from '../../src/replay';
+import { resetSdkMock } from '../mocks/resetSdkMock';
+import { useFakeTimers } from '../utils/use-fake-timers';
+
+useFakeTimers();
+
+describe('Integration | getRecordingMode()', () => {
+  let replay: ReplayContainer;
+  let integration: Replay;
+
+  beforeEach(async () => {
+    ({ replay, integration } = await resetSdkMock({
+      replayOptions: {
+        stickySession: false,
+      },
+      sentryOptions: {
+        replaysSessionSampleRate: 1.0,
+      },
```

Comment:
can we add tests for other combinations of the sample rates? e.g. both 0, onError 1 (sample 0), both 1